src/pyramid/view.py

```diff
@@ -216,6 +217,13 @@ def __init__(self, **settings):
             if settings.get('context') is None:
                 settings['context'] = settings['for_']
         self.__dict__.update(settings)
+        self._get_info()
+
+    def _get_info(self):
+        frame = sys._getframe(2)
```

Comment:
I'm wondering if instead of using `sys._getframe(2)` directly if we shouldn't instead use

```
inspect.stack()[2]
```

instead to grab the frame, then there is no reason to use an underscore method on `sys`.